Documentation/proposals/202305-scrapeclasses.md

```diff
@@ -0,0 +1,189 @@
+
+## Scrape Classes
+
+* **Owners:**
+  * `eronwright`
+
+* **Related Tickets:**
+  * https://github.com/prometheus-operator/prometheus-operator/issues/4121
+  * https://github.com/prometheus-operator/prometheus-operator/issues/3922
+
+This proposal aims to introduce the concept of _scrape classes_, to enable users to leverage scrape configuration data 
+that is provided by the administrator. For example, to allow pod monitors and probes to safely use
+unsafe TLS configuration fields (e.g. `keyFile`). 
+
+## Why
+
+Today, there are some scenarios that are difficult to support in a safe way, notably the need to reference 
+a local TLS file in a scrape configuration. For example, to scrape a pod in an Istio mesh with strict mTLS would require that
+the scrape configuration use a TLS keyfile that is provided by the Prometheus server's Istio sidecar. 
+See [Istio documentation](https://istio.io/latest/docs/ops/integrations/prometheus/#tls-settings) for more details.
+
+Another motivation is to improve feature parity amongst the monitor resources. The `PodMonitor` and `Probe` resources
+aren't at parity with `ServiceMonitor` because the latter allows for unsafe TLS settings.
+
+A minor motivation is to decouple the monitor spec from low-level infrastructure details like TLS certificates.
+
+### Pitfalls of the current solution
+
+The only known solution for use cases where you'd need to use unsafe TLS settings is to use `additionalScrapeConfig`. 
+The downside is obviously the loss of a great feature of the Prometheus Operator. The monitor resources make it possible 
+to compose the scrape configurations in a Kubernetes way.
+
+## Goals
+
+- Allow for the administrator to define a named, reusable scrape configuration snippet, including unsafe elements such as file references.
+- Allow for a user to select a configuration snippet by name in their probe/podmonitor/servicemonitor endpoint configuration.
+- Avoid giving the user the ability to use arbirary files within the Prometheus pod.
+
+### Audience
+
+- Users who serve Prometheus as a service and want to give their customers autonomy in defining monitors.
+- Users who want to scrape pods within an Istio mesh.
+
+## How
+
+The proposed solution is to introduce a notion of a  _scrape class_, akin to a Kubernetes [storage class](https://kubernetes.io/docs/concepts/storage/storage-classes/). 
+A scrape class defines TLS settings (and possibly other settings in future) to be applied to all scrape endpoints of that class. 
+
+When defining a probe/podmonitor/servicemonitor, a user may optionally assign a class to each endpoint. 
+A default class may be defined by the administrator.
+
+Class names are assumed to be installation-specific. In practice, some common class names like `istio-mtls` are likely to emerge.
+
+### Prometheus Resource
+It is proposed that the `Prometheus` resource contain a new section for defining scrape classes.
```

Comment:
Is there a reason for it not to be its own CRD? That way ScrapeClasses can be managed separately from Prometheus itself, in the case where we have more settings than `tlsConfig` inside.

Reply:
Are you advocating for the `ScrapeClass` resource as outlined in alternatives?

Reply:
https://github.com/prometheus-operator/prometheus-operator/pull/5572/files#r1199707802